Code-4-Community/lucys-love-bus-backend
# Added src to announcements

Added an image_src column to the database, and a corresponding imageSrc field to the Announcements DTO, and the create + get announcement requests. Note: the imageSrc is _optional_ for the create announcement request, since the user does not need to specify an image for an announcement. 

## Why

This will now enable users to upload/view announcements with images.

What benefit does this bring to the end user? Or, what benefit does this bring to developers 
working in the codebase?

Users will now get a better idea of an announcement with an image describing it.

## This PR

Added a sql migration file to add an image_src (string) column to our announcements table. Then, I added a string src field to the announcements DTO, matching the new table. I added an imageSrc field to the `PostAnnouncementRequest` object, which can be null. 

## Screenshots

Creating an announcement _with_ an image:
![image](https://user-images.githubusercontent.com/46979800/108939033-139c3480-761f-11eb-8118-bd96d8e1cfa3.png)

Creating an announcement _without_ an image:
![image](https://user-images.githubusercontent.com/46979800/108939072-2151ba00-761f-11eb-8404-c8184d59ca86.png)

Getting announcement (now you can see the newly added imageSrc field):
![image](https://user-images.githubusercontent.com/46979800/108939116-33cbf380-761f-11eb-8e7e-9709fe84c1df.png)

## Verification Steps

What steps did you take to verify your changes work? These should be clear enough for someone to be 
able to clone the branch and follow the steps themselves. 

I added announcements to the table with and without an imageSrc value. I added JUnit tests to create announcements (event specific and non event-specific) with images. I also added JUnit tests to get announcements that had images and also that didn't. Furthermore, I made test requests using Postman to verify the correct functionality.
